### DISCUSSION TITLE
How do build a sample client?

### DISCUSSION THREAD

**Discussion by kimcoldwater** [FreeRDP/FreeRDP#8093] [Q&A] [answered]
hello.

I'm going to create a Linux headless client that doesn't need the $DISPLAY environment variable to implement RDP functionality on the web.

I know how to customize the `client/sample`, but I don't know how to build it.

Failed to run `cmake ..` on path to client/sample.

However, the following error came out.

```
CMake Error at common/CMakeLists.txt:93 (export_complex_library):
  Unknown CMake command "export_complex_library".

```

I wasn't familiar with how to use cmake, so I spent a few days looking at how to use it.

I tried to understand it by comparing it with other client examples.

But it's not easy to solve.

I know using `CMakeLists.txt` , but I don't know how to configure the script.

Please give me some advice for sample build.

Thank you for reading the question.

**Comment by mfleisz** [comment, marked as answer]
You have to point cmake to the root `CMakeLists.txt` file in order to work correctly. Don't use one of the `CMakeLists.txt` files in one of the sub-directories to start your build. This is not how cmake is intended to work.

In order to build the sample client you have to set `WITH_SAMPLE=ON`

**Reply by kimcoldwater** [reply]
Thank you for answer.

If so, are you saying that you only need to turn on the with sample option in the build script for the existing client?

cmake -GNinja -DCHANNEL_URBDRC=ON -DWITH_DSP_FFMPEG=ON -DWITH_CUPS=ON -DWITH_PULSE=ON -DWITH_FAAC=ON -DWITH_FAAD2=ON -DWITH_GSM=ON -B<your build directory> -H<your FreeRDP source directory>


If I add -DWITH_SAMPLE=ON to these existing x11 build methods, can I build the sample client?

**Comment by mfleisz** [comment]
Exactly - it will produce a `sfreerdp` binary from the code in `client/sample` in the output folder.

**Comment by rcarmo** [comment]
How would that work to build the Mac SDL client? I have `cmake`, `openssl`, etc. all installed via `brew`, but cannot figure out the right incantation.
